### PR TITLE
[Fix Scala Warnings] - Add explicit types to implicits

### PR DESCRIPTION
**Pull request by rupertbates**
## What are you doing in this PR?
This PR is the first of potentially several to fix Scala warnings in the support-frontend codebase, it contains two main changes:
- Add [Scalafix](https://scalacenter.github.io/scalafix/) to the project to enable a number of automated fixes
- Use Scalafix to carry out the [ExplicitResultTypes](https://scalacenter.github.io/scalafix/docs/rules/ExplicitResultTypes.html) fix to add explicit types to all implicit vals in the project

### CODE REVIEW

File: .scalafix.conf [hunk 1]
```diff
@@ -2,3 +2,4 @@ rules = [
  ExplicitResultTypes,
 ]
 ExplicitResultTypes.onlyImplicits = true
+ExplicitResultTypes.skipSimpleDefinitions = false
```
Comment (rupertbates) [inline .scalafix.conf:5]: This config tells Scalafix to only carry out the fix on implicits but to do it even for simple val declarations, which it won't do by default.

File: support-payment-api/src/main/scala/MyApplicationLoader.scala [hunk 2]
```diff
@@ -92,7 +92,7 @@ class MyComponents(context: Context)
       .buildRequestBasedProvider(requestEnvironments)
       .valueOr(throw _)
 
-  implicit val allowedCorsUrl = configuration.get[Seq[String]](s"cors.allowedOrigins").toList
+  implicit val allowedCorsUrl: List[String] = configuration.get[Seq[String]](s"cors.allowedOrigins").toList
```
Comment (johnduffell) [inline support-payment-api/src/main/scala/MyApplicationLoader.scala:95]: implicit `List[String]` is a brave one

**Review by johnduffell — APPROVED** [review]
looks good, not as much as I expected to see though

**Comment by rupertbates** [comment]
> looks good, not as much as I expected to see though

Yeah I thought there might be more. This is just implicits though, there will be a lot more for public members

**Pull request merged**